Runtime/Scripts/AudioStreamTrack.cs

```diff
@@ -66,32 +117,72 @@ public void Dispose()
                     WebRTC.DestroyOnMainThread(m_clip);
                 }
                 m_clip = null;
-                m_recvBufs.Dispose();
+                m_recvBufs.Clear();
             }
 
-            internal void SetData(float[] data)
+            internal AudioSource FindAttachedAudioSource()
             {
-                m_recvBufs.TryAdd(data);
+                foreach (var audioSource in GameObject.FindObjectsOfType<AudioSource>())
+                {
+                    if (audioSource.clip != null && audioSource.clip.name == m_clip.name)
+                    {
+                        return audioSource;
+                    }
+                }
+                return null;
             }
 
-            internal void OnAudioRead(float[] data)
+            internal void WriteToAudioClip(int numOfFrames = 1)
             {
-                if (m_recvBufs == null || m_clip == null)
+                int baseOffset = (int)(m_bufInfo.BufferPosition % m_clip.samples);
+                int writtenSamples = 0;
+
+                while (numOfFrames-- > 0)
                 {
-                    return;
+                    writtenSamples += WriteBuffer(
+                        m_recvBufs.Count > 0 ? m_recvBufs.Dequeue() : null,
+                        baseOffset + writtenSamples);
+                }
+
+                m_bufInfo.BufferPosition += writtenSamples;
+
+                int WriteBuffer(float[] data, int offset)
+                {
+                    data ??= new float[m_bufInfo.SamplesPer10ms * m_clip.channels];
+                    m_clip.SetData(data, offset % m_clip.samples);
+                    return data.Length / m_clip.channels;
                 }
+            }
+
+            internal void SetData(float[] data)
+            {
+                m_recvBufs.Enqueue(data);
 
-                int remain = data.Length;
-                while (remain > 0)
+                if (m_recvBufs.Count >= AudioBufferTracker.NumOfFramesForBuffering && m_bufferReady == false)
```

Comment:
```suggestion
                if (m_recvBufs.Count >= AudioBufferTracker.NumOfFramesForBuffering && !m_bufferReady)
```

Reply:
It's fixed from https://github.com/Unity-Technologies/com.unity.webrtc/pull/565/commits/304c8508629f086cd72d70c4adba064cebb2fc3e